packages/lisk-transactions/src/base_transaction.ts

```diff
@@ -233,10 +233,20 @@ export abstract class BaseTransaction {
 	}
 
 	public validate(): TransactionResponse {
-		const errors = [...this._validateSchema(), ...this.validateAsset()];
+		const validateAssetErrors = this.validateAsset();
+		if (validateAssetErrors.length > 0) {
+			return createResponse(this.id, validateAssetErrors);
+		}
+
+		const validateSchemaErrors = this._validateSchema();
```

Comment:
i thought we are not going to split this? what is the purpose of separating this?
I think it's more useful to know all the invalid field (ie: base schema and asset schema validation result together)

Reply:
`_validateSchema` calls `toJSON` which in turn calls `assetToJSON`.  If validation of `asset` fails, calling `assetToJSON` throws an error. Therefore we separate here.